### DISCUSSION TITLE
[Bug] data urls in nft's result in "Invalid Hash" over the NFT in NFT tab of GUI, gallery & detail views

### DISCUSSION THREAD

**Discussion by Anarkoic** [Chia-Network/chia-blockchain-gui#2560] [Ideas]
### What happened?

sample nft

https://mintgarden.io/nfts/nft1tyh9vflg9e9jgjmc3z48szqkc30qexzyrfcsqruq5pul90n2qzaq2dcdex

the nft uses data: urls in both uris and meta_uris

and the 

### Version

2.5.0

### What platform are you using?

Linux

### What ui mode are you using?

GUI

**Comment by BrandtH22** [comment]
Hey @Anarkoic , this is intended behavior as CHIP 5 (onchain metadata chip) requires a URI and not base64 encoding be used in the URI field.

One would need to create an extension or replacement CHIP to have onchain base64 encoding adopted as a standard (though I must mention that I do not recommend this, there are other ways to secure data via the chain that will not bloat the blockchain DB like using datalayer)

**Comment by Anarkoic** [comment]
Thanks, in reviewing CHIP 5 (https://github.com/Chia-Network/chips/blob/main/CHIPs/chip-0005.md), 

"Data URL's" appear to be URI's according to what I've read

Additionally for the data uri in CHIP 5, there is specifically inclusion of "Decentralized storage link, such as IPFS, Arweave, Storj, etc", which would appear to include data urls, which are decentralized storage on the Chia blockchain.

I'd be happy to discuss this by voice if that'd be productive.

Here is the IETF's RFC 2397 spec, 'The "data" URL scheme', which includes base 64 encoding, as linked to from Chordinals.com:
https://www.rfc-editor.org/rfc/rfc2397

**Comment by BrandtH22** [comment]
Converting this to an enhancement request

**Comment by Anarkoic** [comment]
Thank you, this seems more like a bug than an enhancement, imho. The current behavior gives an incorrect error message that there's an "Invalid Hash", when there is not an Invalid Hash. That is separate from adding additional functionality, imho.

**Reply by BrandtH22** [reply]
Hey @Anarkoic , I have confirmed with the team that this is out of scope for the initial integration of the hash validations and inclusion of base64 images for validating hashes is an enhancement request. 

This means our team will work on it as time permits but cannot provide any estimates as to when it might be integrated. If you and team are able to provide a PR for this functionality that will likely expedite the integration process.

Thank you for your understanding and make it an excellent day 🌱

**Comment by Anarkoic** [comment]
thank you
